### ISSUE TITLE
ZKRQ payloads occasionally missing name fields

### ISSUE THREAD

**Issue by tonymke** [tonymke/testrattingcapitals.com#6]
A dozen or so kills a day coming across ZKRQ will have various name attributes missing. zKillboard itself will be able to parse the names on its own - there are several workarounds available.

Look at the ZKRQ source and determine if it's fixable in it. Otherwise implement a workaround and bug CCP.

Logs of event happening:

```
2017-07-26 16:35:03,160 - testrattingcapitals - ERROR - 63704198-ALL service - kill has no shipType or shipName. This will fail. full_response:
{"package": {"killID": 63704198, "killmail": {"solarSystem": {"id": 30004049}, "killID": 63704198, "killTime": "2017.07.26 16:05:15", "attackers": [{"alliance": {"id": 99005805}, "shipType": {"id": 17715}, "corporation": {"id": 98043801}, "character": {"id": 90429215}, "weaponType": {"id": 27387}, "finalBlow": true, "securityStatus": 2.1, "damageDone": 13248}], "attackerCount": 1, "victim": {"alliance": {"id": 99007476}, "damageTaken": 13248, "items": [{"singleton": 0, "itemType": {"id": 2048}, "flag": 14, "quantityDestroyed": 1}, {"singleton": 0, "itemType": {"id": 12805}, "flag": 28, "quantityDestroyed": 69}, {"singleton": 0, "itemType": {"id": 12346}, "flag": 31, "quantityDropped": 1}, {"singleton": 0, "itemType": {"id": 12805}, "flag": 31, "quantityDestroyed": 69}, {"singleton": 0, "itemType": {"id": 22985}, "flag": 5, "quantityDropped": 1000}, {"singleton": 0, "itemType": {"id": 12805}, "flag": 5, "quantityDestroyed": 600}, {"singleton": 0, "itemType": {"id": 2456}, "flag": 87, "quantityDestroyed": 5}, {"singleton": 0, "itemType": {"id": 1999}, "flag": 15, "quantityDestroyed": 1}, {"singleton": 0, "itemType": {"id": 17559}, "flag": 21, "quantityDestroyed": 1}, {"singleton": 0, "itemType": {"id": 2605}, "flag": 16, "quantityDestroyed": 1}, {"singleton": 0, "itemType": {"id": 12346}, "flag": 27, "quantityDestroyed": 1}, {"singleton": 0, "itemType": {"id": 12805}, "flag": 27, "quantityDestroyed": 69}, {"singleton": 0, "itemType": {"id": 12346}, "flag": 28, "quantityDropped": 1}, {"singleton": 0, "itemType": {"id": 10190}, "flag": 11, "quantityDestroyed": 1}, {"singleton": 0, "itemType": {"id": 22977}, "flag": 5, "quantityDestroyed": 1400}, {"singleton": 0, "itemType": {"id": 22983}, "flag": 5, "quantityDestroyed": 1000}, {"singleton": 0, "itemType": {"id": 3244}, "flag": 22, "quantityDropped": 1}, {"singleton": 0, "itemType": {"id": 8419}, "flag": 19, "quantityDestroyed": 1}, {"singleton": 0, "itemType": {"id": 12076}, "flag": 20, "quantityDestroyed": 1}, {"singleton": 0, "itemType": {"id": 12346}, "flag": 29, "quantityDropped": 1}, {"singleton": 0, "itemType": {"id": 28668}, "flag": 5, "quantityDropped": 100}, {"singleton": 0, "itemType": {"id": 12805}, "flag": 30, "quantityDestroyed": 69}, {"singleton": 0, "itemType": {"id": 12805}, "flag": 29, "quantityDestroyed": 69}, {"singleton": 0, "itemType": {"id": 12346}, "flag": 30, "quantityDestroyed": 1}, {"singleton": 0, "itemType": {"id": 31360}, "flag": 92, "quantityDestroyed": 1}, {"singleton": 0, "itemType": {"id": 12801}, "flag": 5, "quantityDropped": 1000}, {"singleton": 0, "itemType": {"id": 31790}, "flag": 93, "quantityDestroyed": 1}, {"singleton": 0, "itemType": {"id": 10190}, "flag": 12, "quantityDestroyed": 1}, {"singleton": 0, "itemType": {"id": 1248}, "flag": 13, "quantityDestroyed": 1}, {"singleton": 0, "itemType": {"id": 31790}, "flag": 94, "quantityDestroyed": 1}], "character": {"id": 1376482246}, "shipType": {"id": 17722}, "corporation": {"id": 98466999}, "position": {"y": -686413593373.53, "x": 3709630179216, "z": -1013703715628.7}}, "war": {"id": 0}}, "zkb": {"locationID": 40256396, "hash": "a200e9332bb0b9af2be7aa08726717711fb38a1d", "fittedValue": 311749018.97, "totalValue": 317101496.97, "points": 51, "npc": false, "href": "https://crest-tq.eveonline.com/killmails/63704198/a200e9332bb0b9af2be7aa08726717711fb38a1d/"}}}


2017-07-26 16:35:03,160 - testrattingcapitals - ERROR - unhandled exception
Traceback (most recent call last):
File "/app/testrattingcapitals/bin/daemon.py", line 51, in main
processing_loop()
File "/app/testrattingcapitals/bin/daemon.py", line 39, in processing_loop
tracked_kill_service.add(label, kill)
File "/app/testrattingcapitals/tracked_kill_service.py", line 19, in add
tk = convert_zk_response_to_tracked_kill(tracking_label, zk_response)
File "/app/testrattingcapitals/tracked_kill_service.py", line 70, in convert_zk_response_to_tracked_kill
tk.ship_name = pk['killmail']['victim']['shipType']['name']
KeyError: 'name'
```

**Comment by tonymke** [comment]
Workaround - https://developers.eveonline.com/resource/resources

**Comment by tonymke** [comment]
Character names can also be missing.
Let's skip persisting any name fields from zkrq

`2017-07-27 10:45:04,205 - testrattingcapitals - DEBUG - Processing loop tick
2017-07-27 10:45:04,206 - testrattingcapitals - DEBUG - Service querying zkrq
2017-07-27 10:45:04,206 - testrattingcapitals - DEBUG - Repository querying zkrq
2017-07-27 10:45:04,723 - testrattingcapitals - INFO - 63717924 zkrq returned
2017-07-27 10:45:04,724 - testrattingcapitals - DEBUG - 63717924 processing
2017-07-27 10:45:04,724 - testrattingcapitals - DEBUG - 63717924 processor DEPLOYMENT_BAD_DRAGON REJECT - wrong alliance_id
2017-07-27 10:45:04,724 - testrattingcapitals - DEBUG - 63717924 processor RATTING_CAPITAL REJECT - wrong alliance_id
2017-07-27 10:45:04,724 - testrattingcapitals - DEBUG - 63717924 processor VNI REJECT - wrong alliance_id
2017-07-27 10:45:04,724 - testrattingcapitals - DEBUG - ? processor ALL ACCEPT
2017-07-27 10:45:04,725 - testrattingcapitals - INFO - 63717924 processed. Labels attached: {'ALL'}
2017-07-27 10:45:04,725 - testrattingcapitals - ERROR - unhandled exception
Traceback (most recent call last):
  File "/app/testrattingcapitals/bin/daemon.py", line 51, in main
    processing_loop()
  File "/app/testrattingcapitals/bin/daemon.py", line 39, in processing_loop
    tracked_kill_service.add(label, kill)
  File "/app/testrattingcapitals/tracked_kill_service.py", line 19, in add
    tk = convert_zk_response_to_tracked_kill(tracking_label, zk_response)
  File "/app/testrattingcapitals/tracked_kill_service.py", line 71, in convert_zk_response_to_tracked_kill
    tk.character_name = pk['killmail']['victim']['character']['name']
KeyError: 'name'
`

**Comment by tonymke** [comment]
Resolved by 785356b
